src/http_tracker_connection.cpp

```diff
@@ -204,6 +204,30 @@ namespace libtorrent
 					// source IP to determine our origin
 					url += "&ip=" + print_address(m_ses.listen_address());
 				}
+
+#if TORRENT_USE_IPV6
+				if (!tracker_req().ipv6 != address_v6() && !i2p)
```

Comment:
Is the the `!` before `tracker_req()` a typo? This line does not build with this `!` .